File: src/main/java/ev/projects/FactoryDisplay.java
package ev.projects;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class FactoryDisplay {
    private final Center center;
    private final List<Tile> tiles;

    public FactoryDisplay(Center center, List<Tile> tiles) {
        this.center = center;
        this.tiles = new ArrayList<>(tiles);
    }

    public int giveTiles(Tile tile) {
        tiles.stream().filter(t -> !t.equals(tile)).forEach(center::addTile);
        long givenTiles = tiles.stream().filter(t -> t.equals(tile)).count();
        tiles.clear();
        return (int) givenTiles;
    }

    public List<Tile> tiles() {
        return tiles;
    }

    public void clear() {
        tiles.clear();
    }

    @Override
    public String toString() {
        if (isEmpty()) {
            return "Empty";
        }
        return Tile.printedTiles(tiles);
    }

    public Map<String, Integer> jsonObject() {
        return Tile.groupedTiles(tiles);
    }

    public boolean isEmpty() {
        return tiles.isEmpty();
    }
}

File: python/src/factory_display.py
from src.tile import Tile

class FactoryDisplay:
    def __init__(self, center, tiles):
        self.center = center
        self.tiles = list(tiles)

    def give_tiles(self, tile):
        given_tiles = sum(1 for t in self.tiles if t == tile)
        for t in self.tiles:
            if t != tile:
                self.center.add_tile(t)
        self.tiles.clear()
        return given_tiles

    def get_tiles(self):
        return self.tiles

    def clear(self):
        self.tiles.clear()

    def __str__(self):
        return "Empty" if self.is_empty() else Tile.printed_tiles(self.tiles)

    def json_object(self):
        return Tile.grouped_tiles(self.tiles)

    def is_empty(self):
        return not self.tiles